libyul/backends/wasm/TextTransform.cpp

```diff
@@ -56,9 +61,10 @@ string TextTransform::run(wasm::Module const& _module)
 
 	// allocate one 64k page of memory and make it available to the Ethereum client
 	ret += "    (memory $memory (export \"memory\") 1)\n";
-	yulAssert(_functionIDs.count("main"), "");
-	// export the main function
-	ret += "    (export \"main\" (func $main))\n";
+	for (auto const& f: _module.functions)
+		if (f.name == "main")
+			// export the main function
+			ret += "    (export \"main\" (func $main))\n";
```

Comment:
Function names can't be reused, right?

```suggestion
			{
				ret += "    (export \"main\" (func $main))\n";
				break;
			}
```